core/src/main/java/org/jboss/set/mjolnir/archive/batch/MembershipRemovalBatchlet.java

```diff
@@ -185,12 +194,39 @@ boolean processRemoval(UserRemoval removal) {
             if (configuration.isUnsubscribeUsers()) {
                 try {
                     teamServiceBean.removeUserFromTeams(organization, gitHubUsername);
+                    for (GitHubTeam gitHubTeam : organization.getTeams()) {
```

Comment:
The problem here is that we don't see which teams failed or succeeded to unsubscribe.

Realistically, it may happen that the first team will unsubscribe successfully, second team will fail with an exception, and a third team will therefore not be processed at all. It is desirable that we are able to record this situation correctly in the audit log.

So if we want to keep audit logging at this level, the `teamServiceBean.removeUserFromTeams()` method should be refactored to process only single team so that we can react to the results accordingly. E.g. it can look like:

```
if (configuration.isUnsubscribeUsers()) {
    for (GitHubTeam team: organization.getTeams()) {
        UnsubscribedUserFromTeam unsubscribedUserFromTeam = new UnsubscribedUserFromTeam();
        // set unsubscribedUserFromTeam fields...
        try {
             teamServiceBean.removeUserFromTeam(team, gitHubUsername); // single team
             unsubscribedUsersFromTeam.setStatus(UnsubscribeStatus.COMPLETED);
             em.persist(removal);
        } catch (...) {
             logRepositoryBean.logError(...);
             unsubscribedUsersFromTeam.setStatus(UnsubscribeStatus.FAILED);
             em.persist(removal);
        }
    }
}
```